src/main/java/com/cloudbees/jenkins/support/configfiles/OtherConfigFilesComponent.java

```diff
@@ -0,0 +1,69 @@
+package com.cloudbees.jenkins.support.configfiles;
+
+import com.cloudbees.jenkins.support.api.Component;
+import com.cloudbees.jenkins.support.api.Container;
+import com.cloudbees.jenkins.support.api.FileContent;
+import edu.umd.cs.findbugs.annotations.NonNull;
+import hudson.Extension;
+import hudson.security.Permission;
+import jenkins.model.Jenkins;
+
+import java.io.File;
+import java.io.FilenameFilter;
+import java.util.Collections;
+import java.util.Set;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+/**
+ * Support component for adding xml files to the support bundle.
+ */
+@Extension
+public class OtherConfigFilesComponent extends Component {
+    @NonNull
+    @Override
+    public Set<Permission> getRequiredPermissions() {
+        return Collections.singleton(Jenkins.ADMINISTER);
+    }
+
+    @NonNull
+    @Override
+    public String getDisplayName() {
+        return "Other Jenkins Configuration Files";
+    }
+
+    @Override
+    public void addContents(@NonNull Container container) {
+        Jenkins jenkins = Jenkins.getInstance();
+        if (jenkins != null) {
```

Comment:
unnecessary null-check. Jenkins instance can (almost) never be null. See Stephen's explanation [here](https://github.com/jenkinsci/jenkins/commit/bb7c8fcedbcc9b51c5b1bb5b32810af5ac6b1ffb) e.g. (note: that commit was reverted afterwards for other reasons :))